src/main/java/com/louis/HashSet.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * 
+ */
+package com.louis;
+
+import java.util.HashMap;
+
+/**
+ * Implements a hashset.
+ *
+ * Hashsets implement the set abstract data type.
+ * The set contains unique elements.
+ * A hash function computes the location of an element.
+ * The location is checked to see if it contains the element.
+ * This implementation merely modifies the inbuilt HashMap.
+ *
+ * @author Louis Ashton (louisashton@live.com)
+ *
+ */
+public class HashSet<E> {
+    // Dummy value; represents a value in the backing Map.
+    private static final Object PRESENT = new Object();
+    private final HashMap<E,Object > map = new HashMap<E,Object>();
+
+    /**
+     * Adds an element to the set.
+     *
+     * @param element is the element to be added.
+     * @throws IllegalArgumentException is thrown if element is null.
+     * @return Returns false if the set already contains that value.
+     */
+    public boolean add(E element) {
```

Comment:
I would argue in the general case that a method should either mutate the object or return a value, not both.